native/dlasy2.go

```diff
@@ -0,0 +1,289 @@
+			btmp[0] *= scale
+			btmp[1] *= scale
+		}
+		// Solve the system [u11 u12] [x21] = [ btmp[0] ].
+		//                  [  0 u22] [x22]   [ btmp[1] ]
+		x22 := btmp[1] / u22
+		x21 := btmp[0]/u11 - (u12/u11)*x22
+		if ipiv&0x1 != 0 { // true for ipiv equal to 1 and 3.
+			// The pivot was in the second column, swap the elements
+			// of the solution.
+			x21, x22 = x22, x21
+		}
+		x[0] = x21
+		if n1 == 1 {
+			x[1] = x22
+			xnorm = math.Abs(x[0]) + math.Abs(x[1])
+		} else {
+			x[ldx] = x22
+			xnorm = math.Max(math.Abs(x[0]), math.Abs(x[ldx]))
+		}
+		return scale, xnorm, ok
+	}
+
+	// 2×2 case: op[TL11 TL12]*[X11 X12] + SGN*[X11 X12]*op[TR11 TR12] = [B11 B12].
+	//             [TL21 TL22] [X21 X22]       [X21 X22]   [TR21 TR22]   [B21 B22]
+	//
+	// Solve equivalent 4×4 system using complete pivoting.
+	// Set pivots less than smin to smin.
+
+	smin := math.Max(math.Abs(tr[0]), math.Abs(tr[1]))
+	smin = math.Max(smin, math.Max(math.Abs(tr[ldtr]), math.Abs(tr[ldtr+1])))
+	smin = math.Max(smin, math.Max(math.Abs(tl[0]), math.Abs(tl[1])))
+	smin = math.Max(smin, math.Max(math.Abs(tl[ldtl]), math.Abs(tl[ldtl+1])))
+	smin = math.Max(eps*smin, smlnum)
+
+	var t [4][4]float64
+	t[0][0] = tl[0] + sgn*tr[0]
+	t[1][1] = tl[0] + sgn*tr[ldtr+1]
+	t[2][2] = tl[ldtl+1] + sgn*tr[0]
+	t[3][3] = tl[ldtl+1] + sgn*tr[ldtr+1]
+	if tranl {
+		t[0][2] = tl[ldtl]
+		t[1][3] = tl[ldtl]
+		t[2][0] = tl[1]
+		t[3][1] = tl[1]
+	} else {
+		t[0][2] = tl[1]
+		t[1][3] = tl[1]
+		t[2][0] = tl[ldtl]
+		t[3][1] = tl[ldtl]
+	}
+	if tranr {
+		t[0][1] = sgn * tr[1]
+		t[1][0] = sgn * tr[ldtr]
+		t[2][3] = sgn * tr[1]
+		t[3][2] = sgn * tr[ldtr]
+	} else {
+		t[0][1] = sgn * tr[ldtr]
+		t[1][0] = sgn * tr[1]
+		t[2][3] = sgn * tr[ldtr]
+		t[3][2] = sgn * tr[1]
+	}
+
+	var btmp [4]float64
+	btmp[0] = b[0]
+	btmp[1] = b[1]
+	btmp[2] = b[ldb]
+	btmp[3] = b[ldb+1]
+
+	// Perform elimination.
+	var jpiv [4]int // jpiv records any column swaps for pivoting.
+	for i := 0; i < 3; i++ {
+		var (
+			xmax       float64
+			ipsv, jpsv int
+		)
+		for ip := i; ip < 4; ip++ {
+			for jp := i; jp < 4; jp++ {
+				if math.Abs(t[ip][jp]) >= xmax {
+					xmax = math.Abs(t[ip][jp])
+					ipsv = ip
+					jpsv = jp
+				}
+			}
+		}
+		if ipsv != i {
+			// The pivot is not in the top row of the unprocessed
+			// block, swap rows ipsv and i of t and btmp.
+			t[ipsv], t[i] = t[i], t[ipsv]
+			btmp[ipsv], btmp[i] = btmp[i], btmp[ipsv]
+		}
+		if jpsv != i {
+			// The pivot is not in the left column of the
+			// unprocessed block, swap columns jpsv and i of t.
+			for k := 0; k < 4; k++ {
+				t[k][jpsv], t[k][i] = t[k][i], t[k][jpsv]
+			}
+		}
+		jpiv[i] = jpsv
+		if math.Abs(t[i][i]) < smin {
+			ok = false
+			t[i][i] = smin
+		}
+		for k := i + 1; k < 4; k++ {
+			t[k][i] /= t[i][i]
+			btmp[k] -= t[k][i] * btmp[i]
+			for j := i + 1; j < 4; j++ {
+				t[k][j] -= t[k][i] * t[i][j]
+			}
+		}
+	}
+	if math.Abs(t[3][3]) < smin {
+		t[3][3] = smin
```

Comment:
It is slightly surprising for me that the reference does not set `INFO = 1` here. Above in the 1x2 case they do set it if u22 is too small. Seems like a tiny bug to me.